rest_server/src/controllers/item_controller.js

```diff
@@ -155,11 +155,152 @@ const del = asyncHandler(async (req, res, next) => {
   }
 });
 
+const listTags = asyncHandler(async (req, res, next) => {
+  try {
+    let result = await MarketplaceItem.get(req.params.itemId);
+    if (checkReadPermission(req.userInfo, result)) {
+      result = await MarketplaceItem.getTags(result);
+      if (isNil(result)) {
+        return next(error.createNotFound());
+      } else {
+        res.status(200).json(result);
+      }
+    } else {
+      return next(
+        error.createForbidden(
+          `The access to the item ${req.params.itemId} is forbidden`,
+        ),
+      );
+    }
+  } catch (e) {
+    return next(error.createInternalServerError(e));
+  }
+});
+
+const addTag = asyncHandler(async (req, res, next) => {
+  try {
+    let result = await MarketplaceItem.get(req.params.itemId);
+    if (checkReadPermission(req.userInfo, result)) {
+      result = await MarketplaceItem.addTag(result, req.params.tagId);
+      if (isNil(result)) {
+        return next(error.createNotFound());
```

Comment:
>createNotFound [](http://example.com/codeflow?start=26&length=14)

for a add function, NotFound is a little weird to me, but if you have your reasons, please ignore this comment. :-)